doc/manual/syntax.md

```diff
@@ -0,0 +1,313 @@
+
+To work with boolean values, we have the classic logical operators *AND* (&&), *OR* (||) and *NOT* (!).
+The _AND_ and _OR_ operators are lazy in the evaluation of the second argument. The truth table below gives the results for using these operators:
+
+Examples:
+```
+> true && false
+false
+
+> false || true
+true
+
+> ! true
+false
+```
+
+### Strings
+
+Nickel can work with sequences of characters, or strings. 
+Strings are enclosed by `" ... "` for a single line string or by `m#" ... "#m` for a multiline string.
+They can be concatenated with the operator `++`.
+Strings must be UTF-8 valid.
+
+String interpolation exists in Nickel, via the notation `"#{ < expression that evaluates to a string > }"`.
+
+Examples:
+```
+> "Hello, World!"
+"Hello, World!"
+
+> m#"Well, if this isn't a multiline string?
+Yes it is, indeed it is"#m
+"Well, if this isn't a string?
+Yes it is, indeed it is"
+
+> "Hello" ++ "World"
+"HelloWorld"
+
+> let h = "Hello" in "#{h} World"
+"Hello World"
+
+> let n = 5 in "The number #{n}."
+error: Type error
+
+> let n = 5 in "The number #{strings.from_num n}."
+"The number 5."
+```
+
+## Equality Operators
+
+We have seen that the operators `==` and `!=` works for comparing numbers, but they can actually compare any values together. To be equal, except for the obvious equality on values, elements must have the same type.
+
+Examples:
+```
+> 1 == 1
+true
+
+> 5 == 5.0
+true
+
+> "Hello" == "Hello"
+true
+
+> "Hello" != "World"
+true
+
+> 5 == "Hello"
+false
+
+> true == "true"
+false
+```
+
+
+## Composite values
+
+### List
+A list is a sequence of values. They are delimited by `[` and `]`, and elements are separated with `,`.
+
+Examples:
+```
+[1, 2, 3]
+["Hello", "World"]
+[1, true, "true"]
+[]
+```
+
+Lists can be concatenated with the operator `@`:
+```
+> [1] @ [2, 3]
+[ 1, 2, 3 ]
+```
+
+### Record
+Records are key-value storage, or in Nickel terms, field-value storage. They are delimited by `{` and `}`, and elements are separated by `,`.
+Field-value elements are noted as `field = value`.
+The fields are strings, but can be noted as identifiers if they respect identifiers syntax (internally, they will be stored as strings). Values can be of any type.
+Elements inside a record are unordered.
+Two records can be _merged_ together using the operator `&`. The reader can find more information about merging in the relevant documentation.
+
+Examples:
+```
+{}
+{a = 3}
+{my_id_n5 = "my id number 5", my_id_n4 = }
+{"5" = 5, six = 6}
+```
+
+Accessing a record field can be done using the `.` operator :
+```
+> { a = 1, b = 5 }.a
+1
+
+> { a = 1 }.b
+error: Missing field
+
+> { "1" = "one" }."1"
+"one"
+```
+
+It is possible to write records of records via the *pieces syntax*, where we separate fields by dots:
+```
+> { a = { b = 1 } }
+{ a = { b = 1 } }
+ 
+> { a.b = 1 }
+{ a = { b = 1 } }
+
+> { a.b = 1, a.c = 2, b = 3}
+{ a = { b = 1, c = 2 }, b = 3 }
+```
+
+## Statements
```

Comment:
There is no statement in Nickel, everything is an expression. Control-flow maybe? But `let .. in ..` is arguably not control flow. Maybe just `Constructs`, or something vague and general enough? Blocks?

Reply:
Constructs seems good for me, I'll use it.